tests/utils.go

```diff
@@ -4330,7 +4330,11 @@ func resetToDefaultConfig() {
 	}
 
 	UpdateKubeVirtConfigValueAndWait(KubeVirtDefaultConfig)
-	UpdateCDIConfigMap(CDIInsecureRegistryConfig)
+
+	// On SRIOV we dont have Insecure registries
+	if !IsRunningOnKindInfra() {
```

Comment:
At the current state we only have sriov provider that kind infra based, maybe we should change `IsRunningOnKindInfra` name.

Reply:
i think that since we dont have others, its fine as is

Reply:
We are trying not to use "kind" because it does not play well with some resources we need in kubevirt (e.g. loopback devices).
I do not expect anything to be added beyond SRIOV specific tests.

Changing the name will add the same question on other places where this condition is used. But those may just want to mention that any kind is not testable for them.

If you want something more accurate, then we can detect that the CDI is not defined and therefore skip on that. It can be reasoned that such an approach is more correct as we do not know if the user ran the tests with or without CDI (e.g. on conformance test).

Reply:
Thanks
i would vote to keep it as the PR does- atm please.

Reply:
What is your argument to keep it like this and not check if CDI is deployed?

Reply:
CDI can be deployed but insecure might not exists for example, so it should check not CDI is there
but insecure regs of CDI are there (unless the function is robust for it, i didnt check deeply)
Moroever i am not sure we need this resolution in this PR, i would prefer if possible to do a follow up, directly after my current task.

Reply:
We have the provider (kind), the CDI and the "insecure" parts... The most accurate is the last I guess, but that function name sais `UpdateCDIConfigMap` which does not mention insecure or how it was implemented (that I guess is abstracted away). It does mention CDI, which implies that without it there is no point to call this function.

@ormergi do you have a suggested alternative condition to add here so it will be CDI specific only?

Reply:
```
		UpdateCDIConfigMap(CDIInsecureRegistryConfig)
```
The param does say CDIInsecureRegistryConfig

Reply:
We should change this function name or create another function to indicate we run on sriov cluster, if not now then in a follow up PR.
And most importantly, add a comment that we do not deploy cdi operator on sriov lane, which is why we skip this line.

Reply:
Who cares that you run on one lane or the other? This function assumes CDI exists, we need to condition on that.
Tomorrow another lane will stop using CDI (e.g. CNAO), we do not want to edit this comment or the function/condition.

I was looking for a condition in the code for checking CDI, is there one?

Reply:
```
	currentConfig, err := virtClient.CoreV1().ConfigMaps(flags.ContainerizedDataImporterNamespace).Get(cdiConfig.Name, metav1.GetOptions{})
	PanicOnError(err)
```
At least not here where we get the panic, this is the condition if we want to add it

```
	CDIInsecureRegistryConfig, err = virtClient.CoreV1().ConfigMaps(flags.ContainerizedDataImporterNamespace).Get(insecureRegistryConfigName, metav1.GetOptions{})
	if err != nil && !errors.IsNotFound(err) {
		PanicOnError(err)
	}
```
This is the problem, that it knows it didnt find it and continue using nil CDIInsecureRegistryConfig

Reply:
Please try to use `HasCDI()`.

Reply:
but please see what i wrote, its a bug that HasCDI wont solve

Reply:
What part is not understood exactly? I fail to comprehend here.
We should **not** call and "update" function on something that does not exist... that is why we should check in advance if it exists.

If you find a logical problem with this, please share.

Reply:
i just did found and this is what i am trying to explain please:
Lets say i will use `HasCDI`
1. For kind SRIOV we will be happy
2. For any other lane that do have CDI, but doesnt have CDIInsecureRegistryConfig, the same panic will occour

because
```
CDIInsecureRegistryConfig, err = virtClient.CoreV1().ConfigMaps(flags.ContainerizedDataImporterNamespace).Get(insecureRegistryConfigName, metav1.GetOptions{})
	if err != nil && !errors.IsNotFound(err) {
		PanicOnError(err)
	}
```
will still let CDIInsecureRegistryConfig be nil

and 
```
	currentConfig, err := virtClient.CoreV1().ConfigMaps(flags.ContainerizedDataImporterNamespace).Get(cdiConfig.Name, metav1.GetOptions{})
	PanicOnError(err)
```
Deref a nil ptr (`cdiConfig.Name`)

Reply:
I am not a CDI expect, I do not know what is expected to work with CDI config, secured, insecure or something else.

As this is seen on the SRIOV lane, I do understand that the question about which provider is used is less accurate than asking about CDI existence or not.
I do not want to call functions or run tests that require CDI on the SRIOV lane, I am also not interested at the moment to change the condition in a function we know nothing about and do not owe, saying that we may support or not support something.

Reply:
i will update the PR with my suggestion and lets speak afterwards

Reply:
Updated to use HasCDI,
will open another PR suggesting a bigger resolution

Reply:
created follow-up PR https://github.com/kubevirt/kubevirt/pull/4472